docs/proposals/proposal-bug-1512042.md

```diff
@@ -0,0 +1,80 @@
+# Proposal for [Bug 1512042](https://bugzilla.redhat.com/show_bug.cgi?id=1512042)
+
+## Introduction
+The issue is that when a custom namespace (not openshift) is used for holding the images the broker does not currently give access to that namespace when creating the service account for the sandbox. This cause's the resulting transient namespace to get stuck in an error pulling the image. 
+
+# Problem Description
+The problem is that we want to be able to control how the privileges get escalated into that namespace to see/pull the images. This means that we will need a robust solution, that is not a simple bug fix.
+
+## Implementation Details
+The proposed changes are adding a configuration option to the local_registry. This configuration option will be similar to the auto_escalate feature we already have. The option will allow for checking if the originating user has access to the custom namespace to pull images. This will be the default, and you can specify to auto grant access to the namespace. There will only be two options for the configuration value `CheckAccessToNamespace` and `AutoGrantAccessToNamespace`.
```

Comment:
later you mention other adapters using this. Why would we add a config option only to the `local_registry`?